builtin/cat-file.c

```diff
@@ -494,22 +495,42 @@ static int batch_objects(struct batch_options *opt)
 	save_warning = warn_on_object_refname_ambiguity;
 	warn_on_object_refname_ambiguity = 0;
 
-	while (strbuf_getline(&buf, stdin) != EOF) {
-		if (data.split_on_whitespace) {
-			/*
-			 * Split at first whitespace, tying off the beginning
-			 * of the string and saving the remainder (or NULL) in
-			 * data.rest.
-			 */
-			char *p = strpbrk(buf.buf, " \t");
-			if (p) {
-				while (*p && strchr(" \t", *p))
-					*p++ = '\0';
+	if (opt.porcelain_format == PORCELAIN_FORMAT_PORCELAIN) {
+		while (strbuf_getline_nul(&buf, stdin) != EOF) {
```

Comment:
Doesn't `strbuf_getline_nul()` split at the *first* NUL? If so, then we should just add another `strbuf_getline_nul()` with a *second* strbuf to assign to `data.rest`. Otherwise we will not have read enough data to see the path.

Reply:
:+1: right there are three parts if `--filter` was used.